tools/GenerateNewControlProject.ps1

```diff
@@ -107,4 +107,75 @@ $xml.Save($testAppProject)
 FindAndReplaceInFile ($muxControlsDir + "\dev\Telemetry\RuntimeProfiler.h") "(\s*ProfId_Size.*\s*})" @"
 
         ProfId_$controlName,`$1
-"@
\ No newline at end of file
+"@
+# Randomize class name for multiple use of class in one powershell
+$id = get-random
+
+# We need double backslash for C# strings below
+$cleanMuxControlsDir = $muxControlsDir.Replace("\","\\") + "\\"
+
+echo "$cleanMuxControlsDir"
+
+$assemblies=(
+	"System","EnvDTE","EnvDTE80"
+)
+
+$source="
+using System;
+using EnvDTE;
+using EnvDTE80;
+using System.Collections.Generic;
+namespace SolutionHelper
+{
+	public static class SolutionRegister$id{
+		public static void Main(){
+            Console.WriteLine(`"Started script`");
+            Type dteType = null;
+            try
+            {
+                dteType = Type.GetTypeFromProgID(`"VisualStudio.DTE.16.0`", true);
+            } catch(Exception)
+            {
+                Console.WriteLine(`"You need to install Visual Studio to add projects to the solution`");
+                return;
+            }
+            var dte = (EnvDTE.DTE)System.Activator.CreateInstance(dteType);
+            var sln = (SolutionClass)dte.Solution;
+            Solution2 solution = (Solution2)dte.Solution;
+            Console.WriteLine(`"Got solution class`");
+
+            var solutionNames = new List<string>(){`"MUXControls.sln`",`"MUXControlsInnerLoop.sln`"};
+            foreach(var solutionName in solutionNames)
+            {
+                Console.WriteLine(`"Opening solution: $($cleanMuxControlsDir)`" + solutionName);
+                solution.Open(`"$cleanMuxControlsDir`" + solutionName);
+                Console.WriteLine(`"Opened solution`");
+                var devFolder = solution.Projects.Item(1);
+    
+                // Get correct reference here:
+                Console.WriteLine(`"Get dev folder`");
+                var solutionFolderAsSolutionFolder = (SolutionFolder)devFolder.Object;
```

Comment:
Name is a bit wordy; I might prefer just `solutionFolder`.  You can use the type rather than `var` to avoid having it in the variable name.

Reply:
Yeah that name wasn't great, updated it now.